compiler/src/dotty/tools/dotc/transform/Getters.scala

```diff
@@ -57,7 +57,8 @@ class Getters extends MiniPhase with SymTransformer {
       d.is(PrivateLocal) && !d.owner.is(Trait) && !isDerivedValueClass(d.owner) && !d.is(Flags.Lazy) ||
       d.is(Module) && d.isStatic ||
       d.hasAnnotation(defn.ScalaStaticAnnot) ||
-      d.isSelfSym
+      d.isSelfSym ||
```

Comment:
The problem here is if you have code like this:
```
enum E extends java.lang.Enum with T {
  case Red
}
trait T {
  def Red: E
}
```
Then you need a getter to implement the trait method.

```